src/NovelRT/Audio/AudioService.cpp

```diff
@@ -3,363 +3,349 @@
+  alGenSources(1, &_soundSource);
+  alSourcef(_musicSource, AL_GAIN, 0.75f);
+  alSourcef(_soundSource, AL_GAIN, 0.75f);
+  alSourcef(_musicSource, AL_PITCH, _pitch);
+  alSourcef(_soundSource, AL_PITCH, _pitch);
+
+  return isInitialised;
 }
 
-void AudioService::load(std::string input, bool isMusic) {
-  if (!isMusic)
-  {
-    auto exists = _sounds.find(input);
-    if (exists != _sounds.end()) return;
+ALuint AudioService::readFile(std::string input) {
+  SF_INFO info;
+  info.format = 0;
+  SNDFILE* file = sf_open(input.c_str(), SFM_READ, &info);
 
-    Mix_Chunk* newSound = Mix_LoadWAV(input.c_str());
-    if (newSound != nullptr)
-    {
-      _sounds[input] = newSound;
-    }
-    else
-    {
-      _logger.logError("SDL_Mixer error occurred during load. Error: ", getSDLError());
-    }
+  if (file == nullptr) {
+    _logger.logErrorLine(std::string(sf_strerror(nullptr)));
   }
-  else
-  {
-    auto exists = _music.find(input);
-    if (exists != _music.end()) return;
 
-    Mix_Music* newMusic = Mix_LoadMUS(input.c_str());
-    if (newMusic != nullptr)
-    {
-      _music[input] = newMusic;
-    }
-    else
-    {
-      _logger.logError("SDL_Mixer error occurred during load. Error: ", getSDLError());
-    }
-  }
-}
+  std::vector<uint16_t> data;
+  std::vector<short> readBuffer;
+  readBuffer.resize(4096);
 
-void AudioService::unload(std::string input, bool isMusic) {
-  if (!isMusic)
-  {
-    auto existingSound = _sounds.find(input);
-    if (existingSound == _sounds.end()) return;
+  size_t readSize = 0;
 
-    Mix_FreeChunk(_sounds[input]);
-    _sounds.erase(existingSound);
+  while ((readSize = sf_read_short(file, readBuffer.data(), readBuffer.size())) != 0) {
+    data.insert(data.end(), readBuffer.begin(), readBuffer.begin() + readSize);
   }
-  else
-  {
-    auto existingMusic = _music.find(input);
-    if (existingMusic == _music.end()) return;
 
-    Mix_FreeMusic(_music[input]);
-    _music.erase(existingMusic);
-  }
+  ALuint buffer;
+  alGenBuffers(1, &buffer);
+  alBufferData(buffer, info.channels == 1 ? AL_FORMAT_MONO16 : AL_FORMAT_STEREO16, &data.front(), data.size() * sizeof(uint16_t), info.samplerate);
+  sf_close(file);
+  return buffer;
 }
 
-void AudioService::playSound(std::string soundName, int loops) {
-  auto existingSound = _sounds.find(soundName);
-  if (existingSound == _sounds.end())
-  {
-    load(soundName, false);
-    existingSound = _sounds.find(soundName);
-    if (existingSound == _sounds.end())
-    {
-      return;
-    }
+/*Note: Due to the current design, this will currently block the thread it is being called on.
+  If it is called on the main thread, please do all loading of audio files at the start of
+  the engine (after NovelRunner has been created).
+*/
+std::vector<ALuint>::iterator AudioService::load(std::string input, bool isMusic) {
+  if (!isInitialised) {
+    _logger.logError("Cannot load new audio into memory while the service is uninitialised! Aborting...");
```

Comment:
For the audio branch, I wonder if we should do something similar to the render API, where we have explicit buffers and audio objects.